core/config/config.go

```diff
@@ -61,7 +61,18 @@ func (c *Config) setAttributes(attrs map[string]interface{}) error {
 	if err != nil {
 		return errors.Trace(err)
 	}
-	c.attributes = result.(map[string]interface{})
+
+	// Ensure that the underlying map is of the correct type, otherwise
+	// this can panic.
+	switch result := result.(type) {
+	case map[string]interface{}:
+		c.attributes = result
+	case ConfigAttributes:
+		c.attributes = result
```

Comment:
`ConfigAttributes` is an alias of `map[string]interface{}`. How would go select a case here?